crypto/param_build_set.c

```diff
@@ -101,21 +101,22 @@ int ossl_param_build_set_multi_key_bn(OSSL_PARAM_BLD *bld, OSSL_PARAM *params,
 {
     int i, sz = sk_BIGNUM_const_num(stk);
     OSSL_PARAM *p;
-
+    const BIGNUM *bn;
 
     if (bld != NULL) {
         for (i = 0; i < sz && names[i] != NULL; ++i) {
-            if (!OSSL_PARAM_BLD_push_BN(bld, names[i],
-                                        sk_BIGNUM_const_value(stk, i)))
+            bn = sk_BIGNUM_const_value(stk, i);
+            if (bn != NULL && !OSSL_PARAM_BLD_push_BN(bld, names[i], bn))
```

Comment:
I donh think this change is the right thing to do..

Reply:
It is absolutely required because the semantics of `OSSL_PARAM_BLD_push_BN(bld, param_name, NULL )` is to push a zero unsigned integer param to the param builder. And that semantics was there on 3.0, 3.1 so we cannot easily change it.

Pushing 0 here means that the imported RSA key will have present CRT params but with zeros inside! That any operation with such key still works is very much surprising to me and IMO it does only by chance.